IDE/Espressif/ESP-IDF/test/test_wolfssl.c

```diff
@@ -116,6 +165,473 @@ static void tskAes256_Test(void *pvParam)
+    if(ret == 0 && (ret = mp_mul(&x, &y, &z)) != MP_OKAY) {
+        TEST_FAIL_MESSAGE("mp_mul() failed\n");
+    }
+
+    mp_radix_size(&z, 16, &radixZ_size);
+    bufZ = (char*)XMALLOC(radixZ_size + 1, NULL, DYNAMIC_TYPE_IN_BUFFER);
+    if(bufZ != NULL) {
+        mp_toradix(&z, bufZ, 16);
+        bufZ[radixZ_size] ='\0';
+    }
+
+    if(verbose) {
+        mp_dump("x_value", &x, 1);
+        mp_dump("y_value", &y, 1);
+        mp_dump("z_value", &z, 1);
+
+        mp_radix_size(&x, 16, &radixX_size);
+        mp_radix_size(&y, 16, &radixY_size);
+        radixX_size = MAX(radixX_size, radixY_size);
+
+        buf = (char*)XMALLOC(radixX_size + 1, NULL, DYNAMIC_TYPE_IN_BUFFER);
+        if(buf != NULL) {
+            mp_toradix(&x, buf, 16);
+            buf[radixX_size] ='\0';
+            printf("X : %s ", buf);
+
+            mp_toradix(&y, buf, 16);
+            buf[radixY_size] ='\0';
+            printf("Y : %s ", buf);
+        }
+        if(bufZ != NULL) {
+            printf("Z : %s \n", bufZ);
+        }
+        XFREE(buf, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    }
+
+    TEST_ASSERT_EQUAL_STRING_MESSAGE(strZ, bufZ,
+                                   "mp_mulmod returns an unexptected result.");
+    XFREE(bufZ, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+
+    return 0;
+
+}
+
+int mp_unitest_mulmod(const char* strZ, const char* strX, const char* strY,
+               const char* strM, int verbose)
+{
+    int ret = 0;
+    char* buf;
+    char* bufZ;
+    int radixX_size;
+    int radixZ_size;
+    int radixY_size;
+    int radixM_size;
+
+    mp_int z, x, y, m;
+
+     /* set init mp_int */
+    if((ret = mp_init_multi(&z, &x, &y, &m, NULL, NULL)) != MP_OKAY) {
+        TEST_FAIL_MESSAGE("mp_init failed.\n");
+    }
+    /* set big-numbers into mp_inits */
+    if(ret == 0 && mp_read_radix(&x, strX, 16) != MP_OKAY) {
+        ESP_LOGE(TAG,"failed to set strX into mp_init");
+        return -1;
+    }
+    if(ret == 0 && mp_read_radix(&y, strY, 16) != MP_OKAY) {
+        ESP_LOGE(TAG,"failed to set strY into mp_init");
+        return -1;
+    }
+    if(ret == 0 && mp_read_radix(&m, strM, 16) != MP_OKAY) {
+        ESP_LOGE(TAG,"failed to set strM into mp_init");
+        return -1;
+    }
+    /* Z = X^Y mod M */
+    if(ret == 0 && (ret = mp_mulmod(&x, &y, &m, &z)) != MP_OKAY) {
+        TEST_FAIL_MESSAGE("mp_mulmod() failed\n");
+    }
+
+    mp_radix_size(&z, 16, &radixZ_size);
+    bufZ = (char*)XMALLOC(radixZ_size + 1, NULL, DYNAMIC_TYPE_IN_BUFFER);
+    if(bufZ != NULL) {
+        mp_toradix(&z, bufZ, 16);
+        bufZ[radixZ_size] ='\0';
+    }
+
+    if(verbose) {
+        mp_dump("x_value", &x, 1);
+        mp_dump("y_value", &y, 1);
+        mp_dump("m_value", &m, 1);
+        mp_dump("z_value", &z, 1);
+
+        mp_radix_size(&x, 16, &radixX_size);
+        mp_radix_size(&y, 16, &radixY_size);
+        mp_radix_size(&m, 16, &radixM_size);
+        radixX_size = MAX(radixX_size, MAX(radixY_size, radixM_size));
```

Comment:
Please use our misc.h `max()`. It already supports proper detection of `WOLFSSL_HAVE_MAX`.

To use misc.h you'll need to include the following code at the top of file (below normal wolfSSL headers):

```
#ifdef NO_INLINE
    #include <wolfssl/wolfcrypt/misc.h>
#else
    #define WOLFSSL_MISC_INCLUDED
    #include <wolfcrypt/src/misc.c>
#endif
```

Reply:
Changed to use max() in misc.h Thank you for pointing out.